nautobot_ssot/integrations/aci/jobs.py

```diff
@@ -2,42 +2,35 @@
 
 from django.templatetags.static import static
 from django.urls import reverse
-from diffsync import DiffSyncFlags
-from nautobot.core.settings_funcs import is_truthy
-from nautobot.extras.jobs import BooleanVar, ChoiceVar, Job
+from diffsync.enum import DiffSyncFlags
+from nautobot.dcim.models import Controller, Location
+from nautobot.extras.choices import SecretsGroupAccessTypeChoices, SecretsGroupSecretTypeChoices
+from nautobot.extras.jobs import BooleanVar, Job, ObjectVar, TextVar
 from nautobot_ssot.jobs.base import DataMapping, DataSource
 from nautobot_ssot.integrations.aci.diffsync.adapters.aci import AciAdapter
 from nautobot_ssot.integrations.aci.diffsync.adapters.nautobot import NautobotAdapter
-from nautobot_ssot.integrations.aci.constant import PLUGIN_CFG
+from nautobot_ssot.integrations.aci.diffsync.client import AciApi
 
 name = "Cisco ACI SSoT"  # pylint: disable=invalid-name, abstract-method
 
-aci_creds = {}
-for key in PLUGIN_CFG["apics"]:
-    subkey = key[key.rfind("_") + 1 :].lower()  # noqa: E203
-    aci_creds.setdefault(subkey, {})
-    if "USERNAME" in key:
-        aci_creds[subkey]["username"] = PLUGIN_CFG["apics"][key]
-    if "PASSWORD" in key:
-        aci_creds[subkey]["password"] = PLUGIN_CFG["apics"][key]
-    if "URI" in key:
-        aci_creds[subkey]["base_uri"] = PLUGIN_CFG["apics"][key]
-    if "VERIFY" in key:
-        aci_creds[subkey]["verify"] = is_truthy(PLUGIN_CFG["apics"][key])
-    if "SITE" in key:
-        aci_creds[subkey]["site"] = PLUGIN_CFG["apics"][key]
-    if "STAGE" in key:
-        aci_creds[subkey]["stage"] = PLUGIN_CFG["apics"][key]
-    if "TENANT" in key:
-        aci_creds[subkey]["tenant_prefix"] = PLUGIN_CFG["apics"][key]
-
 
 class AciDataSource(DataSource, Job):  # pylint: disable=abstract-method
     """ACI SSoT Data Source."""
 
-    apic_choices = [(key, key) for key in aci_creds]
-
-    apic = ChoiceVar(choices=apic_choices, label="Select APIC")
+    apic = ObjectVar(
+        model=Controller,
+        queryset=Controller.objects.all(),
+        display_field="name",
+        required=True,
+        label="ACI APIC",
+    )
+    device_site = ObjectVar(
```

Comment:
I think we can get `device_site` from relation : `controller -> controller_device` as default and make this an optional setting so that sync process requires only a controller